cfpb/sbl-frontend
# [Page] Resolve Errors & Warnings

closes #333 
closes #401 

## Note (05/02/24)
The `sbl-filing` branch used is `main` --  latest working commit `0ddf51bd4bd0ab7aefac8cbb02c7797077885620`

branched off [331-filing-routing_phase2-live-data](https://github.com/cfpb/sbl-frontend/pull/406)

## Changes
- feat(Resolve Errors): Single-Field Errors
- feat(Resolve Errors): Multi-Field Errors
- feat(Resolve Errors): Register-Level Errors
- feat(Resolve Errors): Alerts present depending on errors or no errors
- feat(Resolve Errors): Swappable between Step 1 (Syntax Errors) and Step 2 (Logical Errors/Register Errors); navigates to `warnings` on Step 2 (no errors)
- enhancement(Upload): added an `enableLongPolling` boolean to `useGetSubmissionLatest`; default false
- enhancement(Resolve Errors): Implemented 'react-markdown' -- Validation Description
- chore(Resolve Errors): TypeScript handling in `FilingErrors.helpers.ts`
- [chore(Upload): Hide FilingNavButtons if still loading](https://github.com/cfpb/sbl-frontend/pull/423/commits/d380708ee606c249ebd4b552468c4c7a75cb4930)
- feat(Resolve Errors): if a get submission latest fails, there is an Alert that pops up


## Next PRs
- Pagination
- Accessibility pass
- Table styling size adjustments
- Download linking the validation report

## How to Test

1. `yarn install` to install `react-markdown`
2. Select an institution and upload with the csv files provided (Syntax Errors and then Logical/Register Errors)
3. Ensure the right Alert occurs based on the error (or no errors) per segment
4. Verify all proper data is present per field-level section

#### Test CSV's to use
[all-errors-warnings-no-syntax-errors.csv](https://github.com/cfpb/sbl-frontend/files/15182232/all-errors-warnings-no-syntax-errors.csv)
[sblar_syntax_errors_large.csv](https://github.com/cfpb/sbl-frontend/files/15182233/sblar_syntax_errors_large.csv)
[sblar_syntax_errors.csv](https://github.com/cfpb/sbl-frontend/files/15182234/sblar_syntax_errors.csv)

Note: If Syntax Errors are present, then validation for the logical errors/warnings do not occur. Hence, why the test CSV's has either syntax errors or logical errors; not both.

## Screenshots
|Markdown|
|--|
|<img width="602" alt="Screenshot 2024-05-02 at 2 40 05 AM" src="https://github.com/cfpb/sbl-frontend/assets/13324863/c277389e-46c3-4374-8195-740e3b04b193">|

| SingleFieldErrors | Success | Register-Level |
|--|--|--|
|<img width="411" alt="Screenshot 2024-05-01 at 4 44 52 PM" src="https://github.com/cfpb/sbl-frontend/assets/13324863/776da07d-7c9a-4dca-9fe9-24e106ac00e2">|<img width="400" alt="Screenshot 2024-05-01 at 4 45 10 PM" src="https://github.com/cfpb/sbl-frontend/assets/13324863/4682d70c-52dc-40fb-9f8b-a3e0c72d8247">|<img width="403" alt="Screenshot 2024-05-01 at 4 45 45 PM" src="https://github.com/cfpb/sbl-frontend/assets/13324863/c1b6e28e-c6e1-4b1a-902b-b7803aa91f77">|
